File: frontend/src/app/dto/player.ts
import {GameRoom} from './gameRoom';

export class Player {
  id: number;
  gameRoom: GameRoom;
  name: string;
  score; number;

  constructor(args: Player = <Player>{}) {
    this.id = args.id;
    this.gameRoom = args.gameRoom;
    this.name = args.name;
    this.score = args.score;
  }
}

export class PlayerJoinRequest {
  id: number;
  quizcode: string;
  name: string;

  constructor(args: PlayerJoinRequest = <PlayerJoinRequest>{}) {
    this.id = args.id;
    this.quizcode = args.quizcode;
    this.name = args.name;
  }
}

File: src/main/java/com/corelogic/schemaconverter/entity/Player.java
package com.corelogic.schemaconverter.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@javax.persistence.Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Player {

    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "PlayerIdSeq")
    @SequenceGenerator(name = "PlayerIdSeq", sequenceName = "player_id_seq", allocationSize = 1)
    @Id
    private Long id;

    @OneToOne(cascade = CascadeType.ALL)
    private GameRoom gameRoom;

    private String name;
    private int score;

}

